model/channel_search.go

```diff
@@ -26,5 +27,8 @@ func (c *ChannelSearch) ToJson() string {
 func ChannelSearchFromJson(data io.Reader) *ChannelSearch {
 	var cs *ChannelSearch
 	json.NewDecoder(data).Decode(&cs)
+
+	cs.Term = strings.TrimSpace(cs.Term)
```

Comment:
3/5, but I would say that the responsability of this function it should be just unmarshaling, not any extra transformation.

Reply:
I see it more a responsibility of the `app.SearchChannels` and `app.SearchAllChannels` functions because is related about how we interpret the parameters of the search.